pkg/icingadb/history/sla.go

```diff
@@ -1,14 +1,22 @@
 package history
 
 import (
+	"github.com/icinga/icinga-go-library/redis"
+	"github.com/icinga/icinga-go-library/structify"
+	"github.com/icinga/icingadb/pkg/contracts"
+	"github.com/icinga/icingadb/pkg/icingadb/types"
 	"github.com/icinga/icingadb/pkg/icingadb/v1/history"
-	"github.com/icinga/icingadb/pkg/structify"
-	"github.com/icinga/icingadb/pkg/types"
-	"github.com/redis/go-redis/v9"
 	"reflect"
 )
 
-var slaStateStructify = structify.MakeMapStructifier(reflect.TypeOf((*history.SlaHistoryState)(nil)).Elem(), "json")
+var slaStateStructify = structify.MakeMapStructifier(
+	reflect.TypeOf((*history.SlaHistoryState)(nil)).Elem(),
+	"json",
+	func(a any) {
+		if initer, ok := a.(contracts.Initer); ok {
+			initer.Init()
+		}
+	})
```

Comment:
This inline function appears quite a few times. Define it once in a common place?